python/lsst/db/db.py

```diff
@@ -54,547 +41,84 @@
+                 host=None, port=None, db=None, query=None):
         """
-        Drop table <tableName> in database <dbName>.
+        @param logFlag Flag turning on/off SQLAlchemy logging.
 
-        @param tableName  Table name.
-        @param dbName     Database name.
-        @param mustExist  Flag indicating what to do if the database does not exist.
+        @param defFile  File from which default client values are read.
+                        (The same as read_default_file in mysql **kwargs)
 
-        If <dbName> is not set, the current database name will be used. Raise
-        exception if the table does not exist and the mustExist flag is not set to
-        False.
-        """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        try:
-            self.execCommand0("DROP TABLE %s`%s`" % (dbNameStr, tableName))
-        except DbException as e:
-            if e.errCode() == DbException.TB_DOES_NOT_EXIST and not mustExist:
-                self._log.debug("dropTable failed, mustExist is False")
-            else:
-                raise
+        @param user     User to connect as
 
-    def listTables(self, dbName):
-        """
-        Return list of tables in a given database.
-        """
-        cmd = "SELECT TABLE_NAME FROM information_schema.TABLES "
-        cmd += "WHERE TABLE_SCHEMA='%s'" % dbName
-        rows = self.execCommandN(cmd)
-        return [x[0] for x in rows]
+        @param passwd   Password to use
 
-    def isView(self, tableName, dbName=None):
-        """
-        Return True if the table <tableName> is a view, False otherwise.
+        @param host     Host to connect
 
-        @param tableName  Table name.
-        @param dbName     Database name.
+        @param port     TCP/IP port to connect to
 
-        @return boolean   True if the table is a view. False otherwise.
+        @param db       Database to use
 
-        If <dbName> is not set, the current database name will be used.
-        """
-        dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
-        rows = self.execCommandN("SELECT table_type FROM information_schema.tables "
-               "WHERE table_schema=%s AND table_name='%s'" % (dbNameStr, tableName))
-        return len(rows) == 1 and rows[0][0] == 'VIEW'
+        @param query    Query parameter, as defined by SQLAlchemy
 
-    def getTableContent(self, tableName, dbName=None):
-        """
-        Get contents of the table <tableName>.
+        Initializes SQLAlchemy. There are two ways:
+        1. If defFile is not null, it will be used, and values for: user, passwd,
+           host, port and db will be set and passed to SQLAlchemy.
+        2. If defFile is null, the values for user/passwd/host/port/db/query are
+           simply passed to SQLAlchemy.
 
-        @param tableName  Table name.
-        @param dbName     Database name.
+        Raises DbException if defFile does not exist, or it is incomplete.
```

Comment:
For now, I think there's some value in using DbException here in order to help simplify error handling code. When a general configuration mechanism is established, it may be worthwhile to revisit this and use those exceptions instead, since configuration usually occurs only once, often at startup, of an application. When that happens, it might be worthwhile to move DbException to the mysqlUtils.py class, or possibly remove it altogether and use PEP-249 exceptions (or SQLAlchemy exceptions).